conda_smithy/utils.py

```diff
@@ -1,20 +1,332 @@
+    logger.debug("## END CONFIGURATION\n")
+
+    if config["provider"]["linux_aarch64"] == "default":
+        config["provider"]["linux_aarch64"] = ["travis"]
+
+    if config["provider"]["linux_aarch64"] == "native":
+        config["provider"]["linux_aarch64"] = ["travis"]
+
+    if config["provider"]["linux_ppc64le"] == "default":
+        config["provider"]["linux_ppc64le"] = ["travis"]
+
+    if config["provider"]["linux_ppc64le"] == "native":
+        config["provider"]["linux_ppc64le"] = ["travis"]
+
+    if config["provider"]["linux_s390x"] in {"default", "native"}:
+        config["provider"]["linux_s390x"] = ["travis"]
+
+    config["remote_ci_setup"] = _santize_remote_ci_setup(
+        config["remote_ci_setup"]
+    )
+    if config["conda_install_tool"] == "conda":
+        config["remote_ci_setup_update"] = [
+            MatchSpec(pkg.strip('"').strip("'")).name
+            for pkg in config["remote_ci_setup"]
+        ]
+    else:
+        config["remote_ci_setup_update"] = config["remote_ci_setup"]
+
+    if not config["github_actions"]["triggers"]:
+        self_hosted = config["github_actions"]["self_hosted"]
+        config["github_actions"]["triggers"] = (
+            ["push"] if self_hosted else ["push", "pull_request"]
+        )
+
+    # Run the legacy checks for backwards compatibility
+    config = _legacy_compatibility_checks(config, forge_dir)
+
+    # Fallback handling set to azure, for platforms that are not fully specified by this time
+    for platform, providers in config["provider"].items():
+        for i, provider in enumerate(providers):
+            if provider in {"default", "emulated"}:
+                providers[i] = "azure"
+
+    # Set the environment variable for the compiler stack
+    os.environ["CF_COMPILER_STACK"] = config["compiler_stack"]
+    # Set valid ranger for the supported platforms
+    os.environ["CF_MIN_PY_VER"] = config["min_py_ver"]
+    os.environ["CF_MAX_PY_VER"] = config["max_py_ver"]
+    os.environ["CF_MIN_R_VER"] = config["min_r_ver"]
+    os.environ["CF_MAX_R_VER"] = config["max_r_ver"]
+
+    config["package"] = os.path.basename(forge_dir)
+    if not config["github"]["repo_name"]:
+        feedstock_name = os.path.basename(forge_dir)
+        if not feedstock_name.endswith("-feedstock"):
+            feedstock_name += "-feedstock"
+        config["github"]["repo_name"] = feedstock_name
+    config["exclusive_config_file"] = exclusive_config_file
+
+    return config
+
+
+def _get_metadata_from_feedstock_dir(
+    feedstock_directory: Union[str, os.PathLike], forge_config: Dict[str, Any]
+) -> Union[MetaData, RattlerBuildMetaData]:
+    """
+    Return either the conda-build metadata or rattler-build metadata from the feedstock directory
+    based on conda_build_tool value from forge_config.
+    Raises OsError if no meta.yaml or recipe.yaml is found in feedstock_directory.
+    """
+    if forge_config and forge_config.get("conda_build_tool") == RATTLER_BUILD:
```

Comment:
Will this work with a fresh feedstock? Because I am not sure if we will have the right `forge_config`.

Reply:
@nichmor did you see this comment?